helix-term/src/commands.rs

```diff
@@ -2981,7 +2981,7 @@ pub mod insert {
                         for _ in 0..drop {
                             // delete up to `drop` spaces
                             match chars.next() {
-                                Some(' ') => start -= 1,
+                                Some(c) if c.is_whitespace() => start -= 1,
```

Comment:
I think this won't work correctly with whitespace that has a wider width than 1 (eg. space used in Japanese text).

The body of this function should be extracted into `helix-core` and properly unit tested.

Reply:
> I think this won't work correctly with whitespace that has a wider width than 1 (eg. space used in Japanese text).

Well the Japanese text spaces are Ideographic Spaces.
I also filed this bug because i am using a japanese ime quite frequently.

This change has been tested locally, and works.

However i see what you mean, would it maybe make sense to use the `width` variable that is used a bit higher?
```rs
                          let width: usize = fragment
                            .chars()
                            .map(|ch| {
                                if ch == '\t' {
                                    tab_size
                                } else {
                                    // it can be none if it still meet control characters other than '\t'
                                    // here just set the width to 1 (or some value better?).
                                    ch.width().unwrap_or(1)
                                }
                            })
                            .sum();
```